test/infrastructure/docker/internal/docker/machine.go

```diff
@@ -329,13 +329,28 @@ func (m *Machine) PreloadLoadImages(ctx context.Context, images []string) error
 }
 
 // ExecBootstrap runs bootstrap on a node, this is generally `kubeadm <init|join>`.
-func (m *Machine) ExecBootstrap(ctx context.Context, data string, format bootstrapv1.Format) error {
+func (m *Machine) ExecBootstrap(ctx context.Context, data string, format bootstrapv1.Format, version *string, image string) error {
 	log := ctrl.LoggerFrom(ctx)
 
 	if m.container == nil {
 		return errors.New("unable to set ExecBootstrap. the container hosting this machine does not exists")
 	}
 
+	// Get the kind kindMapping for the target K8s version.
+	// NOTE: The kind kindMapping allows to select the most recent kindest/node image available, if any, as well as
+	// provide info about the mode to be used when starting the kindest/node image itself.
+	if version == nil {
```

Comment:
I don't know much about scenarios where a Kubernetes version is not provided, but for sure CAPD cannot work without it, hence the error management here

Reply:
Makes sense. I could see where templates that didn't specify a version before may no longer work, but since this only affects CAPD and all the test templates specify a version already, I imagine that scenario isn't worth addressing.

Reply:
Yup agree. Basically we don't support version-less machines in CAPD (before we had some fallback for that case, but we never used it)